### PR TITLE
add ObsOchestratorStorage for huaweiplugin.

### PR DESCRIPTION
**Pull request by Alice1319**
add ObsOchestratorStorage for huaweiplugin.

### CODE REVIEW

File: services/huaweicloud/src/test/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorageTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        ConfigService configService = new ConfigService();
+
+        ServiceRegistry serviceRegistry = new ServiceRegistry();
+        serviceRegistry.add(configService);
+
+        storage.onRegister(serviceRegistry);
+
+        Assertions.assertEquals(0, storage.services().size());
+
+//        storage.store("test");
```
Comment (iskey) [inline services/huaweicloud/src/test/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorageTest.java:38]: Unused codes should be deleted.

File: services/huaweicloud/src/test/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorageTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        String value = "{\"name\":\"张三\",\"pwd\":123456}";
```
Comment (iskey) [inline services/huaweicloud/src/test/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorageTest.java:40]: plz, avoid using Chinese.

File: services/huaweicloud/src/test/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorageTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        storage.store("kafka-service", "osc-orchestrator-huaweicloud", "state", value);
+
+        Assertions.assertEquals(1, storage.services().size());
+
+        Assertions.assertTrue(storage.exists("kafka-service", "osc-orchestrator-huaweicloud", "state"));
+
+        String state = storage.getKey("kafka-service", "osc-orchestrator-huaweicloud", "state");
+
+        Assertions.assertEquals("{\"name\":\"张三\",\"pwd\":123456}",state);
```
Comment (iskey) [inline services/huaweicloud/src/test/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorageTest.java:50]: plz, avoid using Chinese.

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java [hunk 4]
```diff
@@ -14,36 +35,150 @@ public String name() {
...
+        } catch (Exception e) {
+            throw new IllegalStateException("Can't remove " + sid + " from Obs", e);
+        }
+    }
+
+    public String getObjectKey(String sid, String pluginName, String key){
+        return sid + "__" + pluginName + "__" + key;
+    }
+
+    public void closeObsClient(){
```
Comment (iskey) [inline services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java:175]: How about making it private?

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java [hunk 5]
```diff
@@ -14,36 +35,150 @@ public String name() {
...
+        try {
+            obsClient.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (iskey) [inline services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java:179]: `log.info` is better.

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java [hunk 6]
```diff
@@ -14,36 +35,150 @@ public String name() {
...
     public void remove(String sid) {
+        try {
+            obsClient.deleteObject(obsBucket.getBucketName(),sid);
+            closeObsClient();
+        } catch (Exception e) {
+            throw new IllegalStateException("Can't remove " + sid + " from Obs", e);
+        }
+    }
+
+    public String getObjectKey(String sid, String pluginName, String key){
```
Comment (iskey) [inline services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java:171]: How about making it private?

File: services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/OrchestratorStorage.java [hunk 7]
```diff
@@ -45,6 +55,13 @@ public interface OrchestratorStorage {
      */
     Set<String> services();
 
+    /**
+     * Remove a managed service id from the store.
+     *
+     * @param sid the managed service id.
+     */
+    void remove(String sid, String pluginName, String key);
```
Comment (iskey) [inline services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/OrchestratorStorage.java:63]: Maybe not necessary, we can use `store("")` temporarily.

File: services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/OrchestratorStorage.java [hunk 8]
```diff
@@ -30,6 +30,16 @@ public interface OrchestratorStorage {
...
 
+    /**
+     * Check if a managed service id is present in the store.
+     *
+     * @param pluginName the name of the OrchestratorPlugin.
+     * @param sid        the managed service id.
+     * @param key        the property key to store
+     * @return true if the service is present in the store, false else.
+     */
+    boolean exists(String sid, String pluginName, String key);
```
Comment (iskey) [inline services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/OrchestratorStorage.java:41]: Maybe not necessary, we can use `getKey()` temporarily.

**Comment by Alice1319** [comment]
resolved  all the conversions.

**Comment by Alice1319** [comment]
resolved all the conversions and merge to huaweicloud osc.

File: plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java [hunk 9]
```diff
@@ -1,49 +1,161 @@
...
 import org.apache.karaf.minho.boot.service.ServiceRegistry;
 import org.apache.karaf.minho.boot.spi.Service;
 import org.eclipse.osc.orchestrator.OrchestratorStorage;
 
+@Slf4j
 public class ObsOrchestratorStorage implements OrchestratorStorage, Service {
 
+    private static final String ACCESS_KEY = "HW_ACCESS_KEY";
+    private static final String SECRET_KEY = "HW_SECRET_KEY";
+    private static final String ENDPOINT = "ENDPOINT";
```
Comment (iskey) [inline plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java:25]:
```suggestion
    private static final String ENDPOINT = "HW_OBS_STORAGE_ENDPOINT";
```
For environment variables, we should make them more precise.

File: plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java [hunk 10]
```diff
@@ -1,49 +1,161 @@
...
+    private static final String LOCATION = "LOCATION";
```
Comment (iskey) [inline plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/ObsOrchestratorStorage.java:26]:
```suggestion
    private static final String LOCATION = "HW_OBS_STORAGE_REGION";
```
How about we change it to `HW_OBS_STORAGE_REGION`.

**Review by iskey — CHANGES_REQUESTED** [review]

**Comment by iskey** [comment]
LGTM

**Review by iskey — APPROVED** [review]

**Pull request merged**